### PR TITLE
Build time speed improvements

### PR DESCRIPTION
**Pull request by Markionium**
Fix follow up to https://github.com/OfficeDev/office-ui-fabric-react/pull/4714

### CODE REVIEW

**Comment by Markionium** [comment]
I made some updates based on the suggestions by @bworline, thanks for that! :D

@dzearing @cliffkoh Please give this a try on your machines, CI seems to pass and it works for me locally, but it would be nice to have some others try it so we can hopefully avoid the same thing from last time ;)

File: scripts/tasks/jest.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ module.exports = function (options) {
       `--config ${jestConfigPath}`,
 
       // Run tests in serial (parallel builds seem to hang rush.)
-      `--runInBand`,
+      process.env.TRAVIS ? `--runInBand` : undefined,
```
Comment (cliffkoh) [inline scripts/tasks/jest.js]: The comment probably needs to be updated

Comment (bworline) [reply scripts/tasks/jest.js]: Also, is there a reason to exempt the CI build from parallel jest?

**Comment by cliffkoh** [comment]
Hey Mark, so I checked out your branch, resolved the merge conflicts for you, and am getting intermittent failures. Investigating...

(also, the travis CI build doesn't do a proper CI build at the moment - will fix)

File: scripts/tasks/jest.js (hunk 1 above)
Comment (cliffkoh) [reply scripts/tasks/jest.js]: One other issue is that TRAVIS isn't the only "CI" entity - some builds take place on VSTS (production/hotfix). 

Something we need to fix all-up (might be out of scope for this PR)

**Comment by cliffkoh** [comment]
Just FYI, the intermittent failure in a `rush rebuild`:

```

FAIL src\components\Facepile\Facepile.test.tsx
  ● Test suite failed to run

    jest: failed to cache transform results in: C:/Users/user/AppData/Local/Temp/jest/jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22/b9/ContextualMenuSplitButton_b9446445d11841a0bbf30b4d68e674ac
    Failure message: EPERM: operation not permitted, rename 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\b9\ContextualMenuSplitButton_b9446445d11841a0bbf30b4d68e674ac.1454797134' -> 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\b9\ContextualMenuSplitButton_b9446445d11841a0bbf30b4d68e674ac'
      
      at Object.fs.renameSync (fs.js:766:18)
      at Function.writeFileSync [as sync] (../../common/temp/node_modules/write-file-atomic/index.js:192:8)

FAIL src\components\ComboBox\ComboBox.deprecated.test.tsx
  ● Test suite failed to run

    jest: failed to cache transform results in: C:/Users/user/AppData/Local/Temp/jest/jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22/ee/index_ee2a9c063c7babead5d59d470486261f
    Failure message: EPERM: operation not permitted, rename 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\ee\index_ee2a9c063c7babead5d59d470486261f.853597350' -> 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\ee\index_ee2a9c063c7babead5d59d470486261f'
      
      at Object.fs.renameSync (fs.js:766:18)
      at Function.writeFileSync [as sync] (../../common/temp/node_modules/write-file-atomic/index.js:192:8)

FAIL src\components\pickers\BasePicker.test.tsx
  ● Test suite failed to run

    jest: failed to cache transform results in: C:/Users/user/AppData/Local/Temp/jest/jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22/ee/index_ee2a9c063c7babead5d59d470486261f
    Failure message: EPERM: operation not permitted, rename 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\ee\index_ee2a9c063c7babead5d59d470486261f.2363039949' -> 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\ee\index_ee2a9c063c7babead5d59d470486261f'
      
      at Object.fs.renameSync (fs.js:766:18)
      at Function.writeFileSync [as sync] (../../common/temp/node_modules/write-file-atomic/index.js:192:8)

PASS src\components\KeytipLayer\KeytipTree.test.tsx
FAIL src\components\SwatchColorPicker\SwatchColorPicker.test.tsx
  ● Test suite failed to run

    jest: failed to cache transform results in: C:/Users/user/AppData/Local/Temp/jest/jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22/3c/VerticalDivider_3c3b118cad6c63c0aa52f474634d8287
    Failure message: EPERM: operation not permitted, rename 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\3c\VerticalDivider_3c3b118cad6c63c0aa52f474634d8287.2050187012' -> 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\3c\VerticalDivider_3c3b118cad6c63c0aa52f474634d8287'
      
      at Object.fs.renameSync (fs.js:766:18)
      at Function.writeFileSync [as sync] (../../common/temp/node_modules/write-file-atomic/index.js:192:8)

FAIL src\components\OverflowSet\OverflowSet.test.tsx
  ● Test suite failed to run

    jest: failed to cache transform results in: C:/Users/user/AppData/Local/Temp/jest/jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22/71/ButtonThemes_71b7d67055b58f11ebf7e29245579846
    Failure message: EPERM: operation not permitted, rename 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\71\ButtonThemes_71b7d67055b58f11ebf7e29245579846.2846348837' -> 'C:\Users\user\AppData\Local\Temp\jest\jest-transform-cache-5c2e4fd0f193c7bbb781801fe2c9c6c2-8e87614c47a948c05801cec0cf200a22\71\ButtonThemes_71b7d67055b58f11ebf7e29245579846'
      
      at Object.fs.renameSync (fs.js:766:18)
      at Function.writeFileSync [as sync] (../../common/temp/node_modules/write-file-atomic/index.js:192:8)
```

**Comment by cliffkoh** [comment]
I have made the change to jest.js :) Once CI passes, this is good to merge. 

Thanks Mark for this PR 👏

**Pull request merged**